_posts/2019-03-28-smart-proxy-capabilities-explained.md

```diff
@@ -0,0 +1,205 @@
+
+## Version 2
+
+In Foreman Proxy 1.22 a new `/v2/features` REST endpoint is introduced. This is a data structure which allows to relay a lot more internal information. Because of this, it is require to authenticate using client certificates. Using `$ curl -s https://$(hostname -f):9090/v2/features --cert /etc/foreman-proxy/foreman_ssl_cert.pem --key /etc/foreman-proxy/foreman_ssl_key.pem | jq` we get:
+
+```json
+{
+  "facts": {
+    "http_enabled": false,
+    "https_enabled": false,
+    "settings": {},
+    "state": "disabled",
+    "capabilities": []
+  },
+  "dns": {
+    "http_enabled": false,
+    "https_enabled": false,
+    "settings": {
+      "use_provider": null
+    },
+    "state": "disabled",
+    "capabilities": []
+  },
+  "templates": {
+    "http_enabled": true,
+    "https_enabled": true,
+    "settings": {},
+    "state": "running",
+    "capabilities": []
+  },
+  "tftp": {
+    "http_enabled": false,
+    "https_enabled": true,
+    "settings": {
+      "tftp_servername": null
+    },
+    "state": "running",
+    "capabilities": []
+  },
+  "dhcp": {
+    "http_enabled": false,
+    "https_enabled": false,
+    "settings": {
+      "use_provider": null
+    },
+    "state": "disabled",
+    "capabilities": []
+  },
+  "puppetca": {
+    "http_enabled": false,
+    "https_enabled": false,
+    "settings": {
+      "use_provider": null
+    },
+    "state": "failed",
+    "capabilities": []
+  },
+  "puppet": {
+    "http_enabled": false,
+    "https_enabled": true,
+    "settings": {
+      "puppet_url": "https://puppet.example.com:8140",
+      "use_provider": [
+        "puppet_proxy_puppet_api"
+      ]
+    },
+    "state": "running",
+    "capabilities": []
+  },
+  "bmc": {
+    "http_enabled": false,
+    "https_enabled": false,
+    "settings": {},
+    "state": "disabled",
+    "capabilities": []
+  },
+  "realm": {
+    "http_enabled": false,
+    "https_enabled": false,
+    "settings": {
+      "use_provider": null
+    },
+    "state": "disabled",
+    "capabilities": []
+  },
+  "logs": {
+    "http_enabled": false,
+    "https_enabled": true,
+    "settings": {},
+    "state": "running",
+    "capabilities": []
+  },
+  "httpboot": {
+    "http_enabled": false,
+    "https_enabled": false,
+    "settings": {},
+    "state": "disabled",
+    "capabilities": []
+  }
+}
+```
+
+As can be seen, it is now a mapping of the feature name to another map representing the feauture's state.
+
+`http_enabled` and `https_enabled` are the result of `enabled` in the feature's setting file. That can be `true`, `false`, `http` or `https`. The installer configures most features using `https` for increased security, but some features need to listen on HTTP. `Templates` and `HTTPBoot` are examples of this because connecting clients wouldn't be able to verify HTTPS or not deal with it at all.
```

Comment:
s/That can be/The setting value can be/ or similar as it was not clear right away what you refer to, http_enabled or the setting.